### PR TITLE
[spritz] Extract logic into collaborators, simplify swipe handling

### PR DESCRIPTION
**Pull request by frapontillo**
The logic for the swipe completion is now handled directly by the `ViewPager` listener callbacks, no need to complete the animation anymore, which makes it possible to get rid of the swipe interpolators and default duration.

All logic has been extracted to collaborators, so it can be unit tested in the future.

A new debug activity has been added with a debug animation to test changes live. The debug activity is exported but is not added to the launcher.

Some logs were added to test edge cases when swiping back and forth. They are disabled by default and can be enabled by setting the system property `Spritz` to `DEBUG`.

The README has been updated accordingly.

#### GIF

_record-for-gif

### CODE REVIEW

File: spritz/README.md [hunk 1]
```diff
@@ -41,9 +41,6 @@ After getting your `LottieAnimationView` and your `ViewPager`, just create a `Sp
 ```java
 Spritz spritz = Spritz
                     .with(lottieAnimationView)
-                    .withDefaultSwipeAnimationDuration(300, TimeUnit.MILLISECONDS)
```
Comment (frapontillo) [inline spritz/README.md]: No more needed, the animation will complete the swipe step according to the offset sent by the viewpager listener.

**Comment by takecare** [comment]
no cool gif of the debug activity :(

**Comment by frapontillo** [comment]
@takecare right! Updated.

File: spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java [hunk 2]
```diff
@@ -1,249 +1,86 @@
...
 
 public class Spritz {
 
-    private final LottieAnimationView lottieAnimationView;
     private final List<SpritzStepWithOffset> spritzStepsWithOffset;
-    private final long totalAnimationDuration;
-    private final long defaultSwipeAnimationDuration;
-    private final TimeInterpolator defaultSwipeForwardInterpolator;
-    private final TimeInterpolator defaultSwipeBackwardsInterpolator;
+    private final SpritzCalculator spritzCalculator;
```
Comment (takecare) [inline spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java]: i'm having a bit of an issue with having everything named `SpritzXYZ` (having `NSBlaBlaBla` flashbacks) but i can't come up with much better names atm (i know there's the risk of collision for at least one or two). just wondering if you considered this (or something similar): `PagerAnimation`, `PagerAnimator`, etc.

File: spritz/spritz-android/src/main/java/com/novoda/spritz/SpritzOnPageChangeListener.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    private void onPageIdle(int position) {
+        log(String.format(Locale.ENGLISH, "Page idle, autoplaying for position %d", position));
+        autoPlay(position);
+        spritzPager.setCachedPosition(position);
+    }
+
+    private void log(String message) {
+        if (Log.isLoggable(TAG, Log.DEBUG) || BuildConfig.DEBUG) {
```
Comment (takecare) [inline spritz/spritz-android/src/main/java/com/novoda/spritz/SpritzOnPageChangeListener.java]: \o/

File: spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java (hunk 2 above)
Comment (frapontillo) [reply spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java]: Yeah, I had the same feeling about this. Maybe I'll rename to `Animation` and `AnimationRunner`?

Comment (takecare) [reply spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java]: `SpritzAnimation` ➡️  `Animation` and `SpritzAnimator` ➡️  `AnimationRunner`? sounds good

Comment (frapontillo) [reply spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java]: Renamed in ba041d3f0f6cc7b70501ded73a5fd42b484fc726:
- `SpritzAnimation` to `Animation`
- `SpritzAnimator` to `AnimationRunner`
- `SpritzCalculator` to `ProgressCalculator`
